alica_engine/src/engine/BasicPlan.cpp

```diff
@@ -61,39 +52,25 @@ void BasicPlan::doRun(void* msg)
 
 void BasicPlan::doTerminate()
 {
-    if (_activeRunJobId != -1) {
-        _engine->editScheduler().cancelJob(_activeRunJobId);
-        _activeRunJobId = -1;
-    }
-    if (setTerminatedState()) {
-        return;
-    }
     try {
-        if (_trace) {
-            _trace->setLog({"status", "terminating"});
-        }
         onTerminate();
-        _trace.reset();
     } catch (const std::exception& e) {
         ALICA_ERROR_MSG("[BasicPlan] Exception in Plan-TERMINATE" << std::endl << e.what());
     }
-
-    _execContext.store(nullptr);
 }
 
-void BasicPlan::notifyAssignmentChange(const std::string& assignedEntryPoint, double oldUtility, double newUtility, size_t numberOfAgents)
+void BasicPlan::traceAssignmentChange(const std::string& assignedEntryPoint, double oldUtility, double newUtility, size_t numberOfAgents)
 {
-    if (_engine->getTraceFactory()) {
-        _engine->editScheduler().schedule(std::bind(&BasicPlan::traceAssignmentChange, this, assignedEntryPoint, oldUtility, newUtility, numberOfAgents));
+    if (_engine->getTraceFactory() && getTrace()) {
```

Comment:
IMO if something is worth tracing it is likely worth logging as well at INFO level.
traceAssignmentChange -> logAssignmentChange that both logs in trace and console

Reply:
That's already logged in detail in PartialAssignment::operator<<  .
The jagger-client-cpp also had some flag that it can output things on console automatically in addition to sending them to the collector.